examples_tests/53.Transformations/main.cpp

```diff
@@ -0,0 +1,508 @@
+			clearValues[0].color.float32[2] = 0.1f;
+			clearValues[0].color.float32[3] = 1.f;
+
+			clearValues[1].depthStencil.depth = 0.0f;
+			clearValues[1].depthStencil.stencil = 0.0f;
+
+			info.renderpass = renderpass;
+			info.framebuffer = fbo;
+			info.clearValueCount = 2u;
+			info.clearValues = clearValues;
+			info.renderArea.offset = { 0, 0 };
+			info.renderArea.extent = { WIN_W, WIN_H };
+			cb->beginRenderPass(&info,asset::ESC_INLINE);
+		}
+		// Update instances buffer 
+		{
+			static float current_rotation = 0.0f;
+			current_rotation += dt * 0.005f * SimulationSpeedScale;
+
+			// Update Planets Transformations
+			for(uint32_t i = 0; i < NumInstances; ++i) {
+				auto & solarSystemObj = solarSystemObjectsData[i];
+
+				core::matrix3x4SIMD translationMat;
+				core::matrix3x4SIMD scaleMat;
+				core::matrix3x4SIMD rotationMat;
+				core::matrix3x4SIMD parentMat;
+				
+				translationMat.setTranslation(solarSystemObj.initialRelativePosition);
+				scaleMat.setScale(core::vectorSIMDf(solarSystemObj.scale));
+				
+				{
+					auto rot = current_rotation + 300; // just offset a bit in time
+					rotationMat.setRotation(core::quaternion(0.0f, rot * solarSystemObj.yRotationSpeed, rot * solarSystemObj.zRotationSpeed));
+				}
+
+				if(solarSystemObj.parentIndex > 0u) {
+					auto parentObj = solarSystemObjectsData[solarSystemObj.parentIndex];
+					parentMat = parentObj.matForChildren;
+				}
+
+				solarSystemObj.matForChildren = matrix3x4SIMD::concatenateBFollowedByA(matrix3x4SIMD::concatenateBFollowedByA(parentMat, rotationMat), translationMat); // parentMat * rotationYMat * translationMat
+				instancesData[i].modelMatrix = matrix3x4SIMD::concatenateBFollowedByA(solarSystemObj.matForChildren, scaleMat); // solarSystemObj.matForChildren * scaleMat
+			}
+
+			asset::SBufferRange<video::IGPUBuffer> range;
+			range.buffer = gpuInstancesBuffer;
+			range.offset = 0;
+			range.size = BUF_SZ;
+			device->updateBufferRangeViaStagingBuffer(graphicsQueue, range, instancesData.data());
+		}
+		// draw
+		{
+			// Draw Stuff 
+			for(uint32_t i = 0; i < gpuObjects.size(); ++i) {
+				auto & gpuObject = gpuObjects[i];
+
+				cb->bindGraphicsPipeline(gpuObject.graphicsPipeline.get());
+				cb->pushConstants(gpuObject.graphicsPipeline->getRenderpassIndependentPipeline()->getLayout(), asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(core::matrix4SIMD), viewProj.pointer());
+				cb->drawMeshBuffer(gpuObject.gpuMesh.get());
+			}
+		}
+		cb->endRenderPass();
+		cb->end();
+		
+		CommonAPI::Submit(device.get(), swapchain.get(), cb.get(), graphicsQueue, imageAcquire[resourceIx].get(), renderFinished[resourceIx].get(), fence.get());
+		CommonAPI::Present(device.get(), swapchain.get(), graphicsQueue, renderFinished[resourceIx].get(), imgnum);
+		
+	}
+
+	return 0;
+}
+
+
+// If you see this line of code, i forgot to remove it
+// It forces the usage of NVIDIA GPU by OpenGL
+extern "C" {  _declspec(dllexport) DWORD NvOptimusEnablement = 0x00000001; }
```

Comment:
add to CommonAPI

Reply:
Added to CommonAPI.
Should I remove it from the example now?

Reply:
removed from examples, 
CommonAPI is a header now